### PR TITLE
Add hostname configuration option to choose the hostname/IP to listen on

### LINKED ISSUE #15: Issue with binding on a specific IP
cdp-listend binds only on the IP given by the hostname lookup.
This make it not working on hosts having multiple IPs and being notified by another interface than the one given by the hostname IP lookup.

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #15

### CODE REVIEW

**Comment by stdweird** [comment]
restest this please

**Comment by stdweird** [comment]
@samary rpm https://jenkins1.ugent.be/job/cdp-listend-pr-builder/lastSuccessfulBuild/org.quattor.daemon$cdp-listend/artifact/org.quattor.daemon/cdp-listend/16.8.1-SNAPSHOT/cdp-listend-16.8.1-SNAPSHOT.rpm

File: src/main/scripts/cdp-listend [hunk 1]
```diff
@@ -239,6 +247,7 @@ cdp-listend [OPTIONS]
   --fetch_smear=SECONDS   fetch run smearing time
   --nch_smear=SECONDS     nch run smearing time\n";
   --facility              syslog facility
+  --hostname=fqdn         hostname/IP to listen on (use 0.0.0.0 to listen on all IPs)
```
Comment (ned21) [inline src/main/scripts/cdp-listend]: Listening on 0.0.0.0 is potentially a security risk. I think it was a feature of the old code that it was so restrictive so it was not possible to accidentally start listening on external interfaces. I don't like encoding configuration policy in code (that's what PAN is for!) but I feel reticent about this addition to the help text.

Comment (stdweird) [inline src/main/scripts/cdp-listend]: @ned21 is it sufficient if i remove the 0.0.0.0 from the help text? 

the old code would only listen on whatever hostname is set to, external/public or not. only localhost is not allowed (altough that is also not really valid anymore, people could listen on localhost and have some routing/kernel rules in places that pick up the packets from other interfaces.)

wrt security, in order to solve #13 properly, i'll add optional support for encrypting messages, similar to what ccm-fetch uses with TRUST. that way you could listen on 0.0.0.0; but that's for later.

**Comment by stdweird** [comment]
@ned21 i removed 0.0.0.0 from the help text

**Comment by ned21** [comment]
LGTM. If @samary can confirm the rpm works as expected its good to merge

**Comment by samary** [comment]
@stdweird : I tried and is seems to work. 
Although I couldn't use the notify ant task (seems to use the hostname in the profile to notify, not sure how notify finds it). 
I tested the notification using nc -u $hostname_private 7777 passing ccm TIMESTAMP as argument.
We should maybe add a new attribute in the profile tree to specify the hostname to use for the notify ant task (could also be used to configure cdp-listend as well). I will try to find a way in quattor/configuration-modules-core#944.

**Comment by stdweird** [comment]
@samary you cannot do this via the ant task, but if you add routes on the deploy host like
```
route add -host pubic.host gw private.host
```
(e.g. via quattor :smile: ) it should do the trick for now.

**Comment by samary** [comment]
@stdweird : I don't know how the ant task is finding the hostname to use, any idea ? Is there a way to change the ant task to read another part in the profile (eg. /software/components/cdp/hostname) to have a 1 to 1 configuration (for coherence purpose) ?

**Comment by stdweird** [comment]
@samary it doesn't. there's nothing the in the profile it can use for that.

**Comment by samary** [comment]
@stdweird : Ok, so how notify finds the hostname to use ?

**Comment by stdweird** [comment]
@samary i'd guess the profile name itself is used?

**Comment by stdweird** [comment]
@samary but you could make it use `/software/components/cdp/hostname` if it existed, but then the ant task has to consume/parse the profiles.

**Comment by samary** [comment]
@stdweird : I tried with the profile name but it not used by notify. Will ask to the mailing list. (Making ant reading the profile is way beyond my current understanding of it).

**Comment by jrha** [comment]
Looks like the profile name from `profiles-info.xml` is what is used.

**Comment by samary** [comment]
@jrha : I'm not sure about that. I tried to change by hand the hostname in `profiles-info.xml` but still notifying on the wrong hostname.
It seems this has deeper implication than we thought : AFAIK, Hostname (therefore FULL_HOSTNAME) is set via `object template profile_hostname;` and used by all the quattor chain (profile name, hostname, notify, ...). 
Not being able to specify another hostname to notify make this change a bit useless for us (except by using the routing hack @stdweird proposed but it is not consistent enough for us).
I'll open a separate issue to track our specific requirement.

**Comment by stdweird** [comment]
@samary what might also work is add the public hostnames with the private ips in /etc/hosts
it's less hacky then the direct route

**Comment by jrha** [comment]
LGTM and fixes #15, so can it be merged?
The issue with notifications should probably be tracked on the [scdb-ant-utils](https://github.com/quattor/scdb-ant-utils) repository.

**Comment by ned21** [comment]
Except it does not really fix the reason behind #15. Given it's not usable, do we really want to clutter up our code base with unused code? Or is it a nice to have feature that's worth keeping even if no one is using it? We've deleted a lot of unused code from Quattor over the years... IMHO it seems a shame to add new unused code.

**Comment by jrha** [comment]
It seems like a basic feature that would be nice to have to me, but I see your point.

**Comment by stdweird** [comment]
@ned21 what is missing to make it usable? it's "only" the notifcation mechanism in the ant task, or is there something else i missed?

**Comment by ned21** [comment]
I was going off this quote from @samary  and did not dive into the details:
> Not being able to specify another hostname to notify make this change a bit useless for us

So we might fix something else and then discover another problem further down the stack, etc etc.

**Comment by jrha** [comment]
It's a good point, I'll bump this and think about a top-level issue for the problem.

**Pull request merged**